src/pages/Models/Details/ModelDetails.js

```diff
@@ -32,59 +32,69 @@ import "./_model-details.scss";
-      );
-      if (!found) {
-        delete units[entry[0]];
+
+    const application = filteredData.applications[appName];
+    // remove the units from the application objects that are not
+    // the filter-by app.
+    const subordinateTo = application.subordinateTo || [];
+    Object.keys(filteredData.applications).forEach((key) => {
+      if (key !== appName && !subordinateTo.includes(key)) {
+        filteredData.applications[key].units = {};
       }
     });
-  });
+    // Loop through all of the units of the remaining applications that are
+    // listed in the subordinateTo list. This is done because although
+    // a subordinate is supposed to be installed on each unit, that's not
+    // always the case.
+    subordinateTo.forEach((parentName) => {
+      const units = filteredData.applications[parentName].units;
+      Object.entries(units).forEach((entry) => {
+        const found = Object.entries(entry[1].subordinates).find(
+          (ele) => ele[0].split("/")[0] === appName
+        );
+        if (!found) {
+          delete units[entry[0]];
+        }
+      });
+    });
 
-  // Remove all the machines that the selected application isn't installed on.
-  const appMachines = new Set();
-  for (let unitId in application.units) {
-    const unit = application.units[unitId];
-    appMachines.add(unit.machine);
-  }
-  subordinateTo.forEach((subAppName) => {
-    // this will be the parent of the subordinate and grab the machines from it
-    const parent = filteredData.applications[subAppName];
-    for (let unitId in parent.units) {
-      const unit = parent.units[unitId];
+    // Remove all the machines that the selected application isn't installed on.
+    const appMachines = new Set();
+    for (let unitId in application.units) {
+      const unit = application.units[unitId];
       appMachines.add(unit.machine);
     }
-  });
-  for (let machineId in filteredData.machines) {
-    if (!appMachines.has(machineId)) delete filteredData.machines[machineId];
-  }
+    subordinateTo.forEach((subAppName) => {
+      // this will be the parent of the subordinate and grab the machines from it
+      const parent = filteredData.applications[subAppName];
+      for (let unitId in parent.units) {
+        const unit = parent.units[unitId];
+        appMachines.add(unit.machine);
+      }
+    });
+    for (let machineId in filteredData.machines) {
+      if (!appMachines.has(machineId)) delete filteredData.machines[machineId];
+    }
 
-  // Remove all relations that don't involve the selected application.
-  filteredData.relations = modelStatusData.relations.filter(
-    (relation) => relation.key.indexOf(appName) > -1
-  );
+    // Remove all relations that don't involve the selected application.
+    filteredData.relations = modelStatusData.relations.filter(
+      (relation) => relation.key.indexOf(appName) > -1
+    );
 
-  return filteredData;
+    return filteredData;
+  } else {
```

Comment:
This else isn't necessary because you're either returning at the end of the conditional above or returning `modelStatusData` there is no path in which you would return `undefined` at the end of the function. So you can just explicitly return `modelStatusData` at the end of the function.

Reply:
Good point, done.